### PR TITLE
add: appwrapper pending status and condition check for scale-up

### PR DESCRIPTION
**Pull request by VanillaSpoon**
# Issue link
Closes: https://issues.redhat.com/browse/RHOAIENG-861

# What changes have been made
This pr contains an update to ensure the AW is in a pending state, with a condition of Insufficient resources before scaling up the specified resources for the appwrapper. This ensure resource usage efficiency for the cluster. 

# Verification steps
Steps I have followed to ensure the functionality. On a cluster with 2 m6i.2xlarge worker nodes. 

I provisioned an appwrapper which required more resources than available on the cluster. 

```yaml
    - custompodresources:
      - limits:
          cpu: 10
          memory: 40G
          nvidia.com/gpu: 0
        replicas: 2
        requests:
          cpu: 10
          memory: 40G
          nvidia.com/gpu: 0
      - limits:
          cpu: 10
          memory: 40G
          nvidia.com/gpu: 1
        replicas: 
        requests:
          cpu: 10
          memory: 40G 
          nvidia.com/gpu: 1
```

This ensured the aw would reach pending state with the condition on Insufficient resources, inducing a scale up of the specified resources, as can be seen here: 

ScaleUp

I also provisioned an appwrapper with required resources within the clusters available

```yaml
    - custompodresources:
      - limits:
          cpu: 2
          memory: 8G
          nvidia.com/gpu: 0
        replicas: 1
        requests:
          cpu: 2
          memory: 8G
          nvidia.com/gpu: 0
      - limits:
          cpu: 2
          memory: 8G
          nvidia.com/gpu: 1
        replicas: 2
        requests:
          cpu: 2
          memory: 8G
          nvidia.com/gpu: 1
```

With this the appwrapper was dispatched without the need for scaling up. As can be seen in the following:
NoScaleUp

## Checks
- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [x] Manual tests
   - [ ] Testing is not required for this change

### CODE REVIEW

**Comment by VanillaSpoon** [comment]
Hey @asm582 
I just pushed some changes there regarding the conditions message :)

File: controllers/appwrapper_controller.go [hunk 1]
```diff
@@ -120,17 +120,24 @@ func (r *AppWrapperReconciler) Reconcile(ctx context.Context, req ctrl.Request)
...
-	} else {
-		switch strings.ToLower(r.Config.MachineSetsStrategy) {
-		case "reuse":
-			return r.reconcileReuseMachineSet(ctx, &appwrapper, demandPerInstanceType)
-		case "duplicate":
-			return r.reconcileCreateMachineSet(ctx, &appwrapper, demandPerInstanceType)
+	status := appwrapper.Status.State
+	allconditions := appwrapper.Status.Conditions
+	for _, condition := range allconditions {
+		if status == "Pending" && strings.Contains(condition.Message, "Insufficient") {
```
Comment (asm582) [inline controllers/appwrapper_controller.go:126]: Thanks! I would create a func called:  `func containsInsufficientCondition(allconditions)` that loops through all conditions of an AW. this simplifies the code and helps us check more or less conditions in the future.

**Review by asm582 — COMMENTED** [review]
Please address the review

**Comment by asm582** [comment]
Thanks @VanillaSpoon for the quick turnaround, Can you please add some screenshots of manual tests performed on this PR please?

**Comment by VanillaSpoon** [comment]
Hi @asm582 
I have added screenshots and an explanation of the scale-up testing to the pr description :)

**Comment by asm582** [comment]
Thanks @VanillaSpoon 

LGTM

**Review by asm582 — APPROVED** [review]

**Pull request merged**